guides/hack/56-modules/05-traits.md

```diff
@@ -0,0 +1,44 @@
+Traits have special behavior in Hack when used with modules. 
+
+## Marking traits internal 
+An internal trait can only be used within the module it's defined. Internal traits can have public, protected, or internal methods. 
+
+```hack
+module foo;
+internal trait TFoo {
+    public function foo(): void {}
+    internal function bar(): void {}
+}
+```
+
+```hack
+module bar;
+public class Bar {
+    use TFoo; // error, TFoo is internal
+}
+```
+
+
+
+## Public traits
+Since public traits have "copy-paste" behavior in that their implementations can be copied to any other module, these traits must be treated as if they are outside of any module for safety. Therefore, public traits cannot access any internal symbols or implementations from the module they are defined in. 
```

Comment:
```suggestion
Since public traits have "copy-paste" behavior in that their implementations can be copied to any other module, these traits are treated as if they are outside of any module. Therefore, public traits cannot access any internal symbols or implementations from the module they are defined in. 
```

Rationale makes total sense for Hacklang folks, but less for users :)